pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +813,150 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+// UpdateTTL Call clickhouse-client to send ttl change request
+func (fa *flowAggregator) UpdateTTL(ttl string) {
+	if fa.dbExportProcess == nil {
+		klog.ErrorS(nil, "ClickHouse hasn't been enabled yet")
+		return
+	}
+	if err := fa.dbExportProcess.UpdateTTL(ttl); err != nil {
+		klog.ErrorS(nil, "Error when updating ttl")
+		return
+	}
+}
+
+// updateIfConfigChanges has a watcher to detect if modification happens. Then check whether we should update clickhouse or flow-collector at runtime
+func (fa *flowAggregator) updateIfConfigChanges() {
+
+	configMapPrevious, configMapName, err := flowaggregatorconfig.GetFAConfigMap(fa.k8sClient)
+	if err != nil {
+		klog.ErrorS(err, "Get flow aggregator configmap failed:")
+	}
+	flowAggregatorPreviousData := []byte(configMapPrevious.Data["flow-aggregator.conf"])
+	klog.Info("Watching for flow-aggregator configuration changes")
+
+	watcher, err := fa.k8sClient.CoreV1().ConfigMaps("flow-aggregator").Watch(context.TODO(), metav1.SingleObject(metav1.ObjectMeta{
```

Comment:
The watch always stops after a while (1 hour?, I don't remember exactly how long). I think with the current code, you will not handle that case. You should be able to restart the watch. You can check out https://github.com/antrea-io/antrea/blob/af567666a9ace24373b7e225dfe9255f6063edff/pkg/agent/controller/networkpolicy/networkpolicy_controller.go#L442-L447 to see how we do it in the Antrea Agent.

Reply:
I've restructured this part. Thank you so much!